run-tests can exit as success, even if some tests fail

In pull request #241 , I broke the Makefile-powered tests, and yet, Travis sees tests passing:

```none
Writing code to "Lexer.java"
java -jar ../../lib/java-cup-11b.jar < ycalc.cup
Error: Unable to access jarfile ../../lib/java-cup-11b.jar
make: *** [parser.java] Error 1
./run-tests: line 37: cd: interpreter: No such file or directory
./run-tests: line 38: cd: java: No such file or directory
./run-tests: line 39: cd: simple-maven: No such file or directory
./run-tests: line 40: cd: standalone-maven: No such file or directory
./run-tests: line 41: cd: zero-reader: No such file or directory
./run-tests: line 44: cd: jflex: No such file or directory
The command "./run-tests" exited with 0.
Done. Your build exited with 0.
```

https://travis-ci.org/jflex-de/jflex/jobs/297505025

--- a/jflex/examples/cup/Makefile
+++ b/jflex/examples/cup/Makefile
@@ -1,8 +1,7 @@
 JAVA=java
 JAVAC=javac
 # Root of the project
-# regis Wrong on purpose, I want to see Travis fail
-ROOT=../..
+ROOT=../../..
 JFLEX=$(ROOT)/jflex/bin/jflex
 CUPJAR=$(ROOT)/cup/cup/java-cup-11b.jar
 CUP=$(JAVA) -jar $(CUPJAR) <

--- a/jflex/examples/cup/Makefile
+++ b/jflex/examples/cup/Makefile
@@ -2,8 +2,8 @@ JAVA=java
 JAVAC=javac
 # Root of the project
 ROOT=../../..
-JFLEX=$ROOT/jflex/bin/jflex
-CUPJAR=$ROOT/cup/cup/java-cup-11b.jar
+JFLEX=$(ROOT)/jflex/bin/jflex
+CUPJAR=$(ROOT)/cup/cup/java-cup-11b.jar
 CUP=$(JAVA) -jar $(CUPJAR) <
 CP=.:$(CUPJAR)
 

--- a/jflex/examples/cup/Makefile
+++ b/jflex/examples/cup/Makefile
@@ -1,7 +1,8 @@
 JAVA=java
 JAVAC=javac
 # Root of the project
-ROOT=../../..
+# regis Wrong on purpose, I want to see Travis fail
+ROOT=../..
 JFLEX=$(ROOT)/jflex/bin/jflex
 CUPJAR=$(ROOT)/cup/cup/java-cup-11b.jar
 CUP=$(JAVA) -jar $(CUPJAR) <

--- a/jflex/examples/interpreter/Makefile
+++ b/jflex/examples/interpreter/Makefile
@@ -13,8 +13,8 @@ JAVA=java
 JAVAC=javac
 # Root of the project
 ROOT=../../..
-JFLEX=$ROOT/jflex/bin/jflex
-CUPJAR=$ROOT/cup/cup/java-cup-11b.jar
+JFLEX=$(ROOT)/jflex/bin/jflex
+CUPJAR=$(ROOT)/cup/cup/java-cup-11b.jar
 CUP=$(JAVA) -jar $(CUPJAR)
 CP=.:$(CUPJAR)
 

--- a/jflex/examples/java/Makefile
+++ b/jflex/examples/java/Makefile
@@ -5,8 +5,8 @@
 CUP        = java java_cup.Main -interface <
 # Root of the project
 ROOT       = ../../..
-JFLEX      = $ROOT/jflex/bin/jflex
-CUPJAR     = $ROOT/cup/cup/java-cup-11b.jar
+JFLEX      = $(ROOT)/jflex/bin/jflex
+CUPJAR     = $(ROOT)/cup/cup/java-cup-11b.jar
 CP         = .:$(CUPJAR)
 JAVA       = java
 JAVAC      = javac

--- a/run-tests
+++ b/run-tests
@@ -31,14 +31,21 @@ cd ../..
 
 ## run jflex examples:
 cd jflex/examples
+cd simple-maven && mvn test
+cd ..
+cd standalone-maven && mvn test
+cd ..
 # don't assume byacc/j is installed, just run lexer
-cd byaccj && make clean && make Yylex.java && cd ..
-cd cup && make clean && make && cd ..
-cd interpreter && make clean && make && cd ..
-cd java && make clean && make && cd ..
-cd simple-maven && mvn clean test && cd ..
-cd standalone-maven && mvn clean test && cd ..
-cd zero-reader && make clean && make && cd ..
+cd byaccj && make clean && make Yylex.java
+cd ..
+cd cup && make clean && make
+cd ..
+cd interpreter && make clean && make
+cd ..
+cd java && make clean && make
+cd ..
+cd zero-reader && make clean && make
+cd ..
 
 # also check ant build
 cd jflex && ant gettools build test

--- a/run-tests
+++ b/run-tests
@@ -29,7 +29,9 @@ fi
 cd testsuite/testcases && ../../mvnw test
 cd ../..
 
-## run jflex examples:
+## run jflex examples
+# Each line must end with the test command to make the script exit
+# in case of error (see #242)
 cd jflex/examples
 cd simple-maven && mvn test
 cd ..

